laws/src/main/scala/cats/laws/ApplyLaws.scala

```diff
@@ -26,6 +26,12 @@ trait ApplyLaws[F[_]] extends FunctorLaws[F] with SemigroupalLaws[F] {
 
   def productLConsistency[A, B](fa: F[A], fb: F[B]): IsEq[F[A]] =
     F.productL(fa)(fb) <-> F.map2(fa, fb)((a, _) => a)
+
+  def selectAConsistency[A, B](fab: F[Either[A, B]], ff: F[A => B]): IsEq[F[B]] =
```

Comment:
I don't know if we need this law thinking more on it. If we remove this law, and move all the other non-rigid laws down to Apply or Applicative, I think we can allow selectA == select in a RigidSelective.

This allows users who don't care about rigidity to just pull in Apply/Applicative and use selectA, and users that require rigidity to pull in Selective (or if we must RigidSelective).

To me that is the best way to allow weak and strong versions of select.

Reply:
Because we don't want one operation that's always rigid and another one that's always not.  We need one that is maybe rigid or maybe not.  The canonical example of this is analysis that overapproximates and underapproximates effects.

I think our failure to agree on this point is the root of every other open debate on this PR.